PR: https://github.com/odoo/odoo/pull/233736

From: 50620bb37da348d9dd34d0f6de99c7daac895c14
From: amdi-odoo
Date: 2025-12-18 13:23:36

Breaking data model changes scores: del:8 + add:0, change matches:
-    legend_blocked = fields.Char(related='stage_id.legend_blocked', string='Kanban Blocked')
-    legend_done = fields.Char(related='stage_id.legend_done', string='Kanban Valid')
-    legend_waiting = fields.Char(related='stage_id.legend_waiting', string='Kanban Waiting')
-    legend_normal = fields.Char(related='stage_id.legend_normal', string='Kanban Ongoing')
-    legend_blocked = fields.Char(
-    legend_waiting = fields.Char(
-    legend_done = fields.Char(
-    legend_normal = fields.Char(

Total Changes: 14

[IMP] *: improve stage form view

*: crm, {sale_}project{_sms}, hr_recruitment

Improve the stage form views by unifying every views,
moving fields, hiding sequence field and adding placeholders.

Removing the hr_recruitment custom tooltips feature for the kanban states
as it wasn't really used and to unify with other views.

Adding a color field on hr recruitment stages.

Task-5186979

Part-of: odoo/odoo#233736
Related: odoo/enterprise#98444
Related: odoo/upgrade#8731
Signed-off-by: Warnon Aurélien (awa) <awa@odoo.com>

================================= pseudo patch: =================================

--- a/addons/hr_recruitment/models/hr_applicant.py
+++ b/addons/hr_recruitment/models/hr_applicant.py
@@ -116,14 +116,10 @@ class HrApplicant(models.Model):
     attachment_ids = fields.One2many('ir.attachment', 'res_id', domain=[('res_model', '=', 'hr.applicant')], string='Attachments')
     kanban_state = fields.Selection([
         ('normal', 'In Progress'),
-        ('done', 'Ready for Next Stage'),
+        ('done', 'Ready'),
         ('waiting', 'Waiting'),
         ('blocked', 'Blocked')], string='Kanban State',
         copy=False, default='normal', required=True)
-    legend_blocked = fields.Char(related='stage_id.legend_blocked', string='Kanban Blocked')
-    legend_done = fields.Char(related='stage_id.legend_done', string='Kanban Valid')
-    legend_waiting = fields.Char(related='stage_id.legend_waiting', string='Kanban Waiting')
-    legend_normal = fields.Char(related='stage_id.legend_normal', string='Kanban Ongoing')
     refuse_reason_id = fields.Many2one('hr.applicant.refuse.reason', string='Refuse Reason', tracking=True)
     meeting_ids = fields.One2many('calendar.event', 'applicant_id', 'Meetings')
     meeting_display_text = fields.Char(compute='_compute_meeting_display')

--- a/addons/hr_recruitment/models/hr_recruitment_stage.py
+++ b/addons/hr_recruitment/models/hr_recruitment_stage.py
@@ -26,14 +26,6 @@ class HrRecruitmentStage(models.Model):
         help="If checked, this stage is used to determine the hire date of an applicant")
     rotting_threshold_days = fields.Integer('Days to rot', default=0, help='Day count before applicants in this stage become stale. \
         Set to 0 to disable.  Changing this parameter will not affect the rotting status/date of resources last updated before this change.')
-    legend_blocked = fields.Char(
-        'Red Kanban Label', default=lambda self: _('Blocked'), translate=True, required=True)
-    legend_waiting = fields.Char(
-        'Orange Kanban Label', default=lambda self: _('Waiting'), translate=True, required=True)
-    legend_done = fields.Char(
-        'Green Kanban Label', default=lambda self: _('Ready for Next Stage'), translate=True, required=True)
-    legend_normal = fields.Char(
-        'Grey Kanban Label', default=lambda self: _('In Progress'), translate=True, required=True)
     is_warning_visible = fields.Boolean(compute='_compute_is_warning_visible')
 
     @api.model
